pkg/model/appconfig.go

```diff
@@ -78,11 +82,65 @@ func (b *AppConfig) addToModel(model *BackstageModel, _ bsv1.Backstage) (bool, e
 }
 
 // implementation of RuntimeObject interface
-func (b *AppConfig) updateAndValidate(m *BackstageModel, _ bsv1.Backstage) error {
+func (b *AppConfig) updateAndValidate(m *BackstageModel, backstage bsv1.Backstage) error {
+	err := b.updateDefaultBaseUrls(backstage)
+	if err != nil {
+		return err
+	}
 	updatePodWithAppConfig(m.backstageDeployment, m.backstageDeployment.container(), b.ConfigMap.Name, m.backstageDeployment.defaultMountPath(), "", true, maps.Keys(b.ConfigMap.Data))
 	return nil
 }
 
+// updateDefaultBaseUrls detects if it is running on OCP and tries to set the baseUrl in the default app-config,
+// per the cluster ingress domain and the Route spec.
+// Note that this is purposely done on a best effort basis. So it is not considered an issue if the cluster ingress domain
+// could not be determined, since the user can always set it explicitly in their custom app-config.
+func (b *AppConfig) updateDefaultBaseUrls(backstage bsv1.Backstage) error {
+	if !backstage.Spec.IsRouteEnabled() {
+		return nil
+	}
+	isOcp, err := utils.IsOpenshift()
+	if err != nil {
+		klog.V(1).Infof("could not determine if whether we are running on OCP, skipping setting the default baseUrls: %v", err)
+		return nil
+	}
+	if !isOcp {
+		return nil
+	}
+	host := fmt.Sprintf("%s-%s", RouteName(backstage.Name), backstage.Namespace)
+	appendIngressDomain := true
+	if backstage.Spec.Application != nil && backstage.Spec.Application.Route != nil {
+		// Per the Route spec, if a user specifies both the host and subdomain, the host takes precedence.
+		if backstage.Spec.Application.Route.Host != "" {
+			host = backstage.Spec.Application.Route.Host
+			appendIngressDomain = false
+		} else if backstage.Spec.Application.Route.Subdomain != "" {
+			host = backstage.Spec.Application.Route.Subdomain
+		}
+	}
+	if appendIngressDomain {
+		var domain string
+		domain, err = utils.GetOCPIngressDomain()
+		if err != nil {
+			klog.V(1).Infof("could not get cluster ingress domain, skipping setting the default baseUrls: %v", err)
+			return nil
+		}
+		if domain == "" {
+			klog.V(1).Info("no cluster ingress domain found, skipping setting the default baseUrls")
+			return nil
+		}
+		host += fmt.Sprintf(".%s", domain)
+	}
+	baseUrl := fmt.Sprintf("https://%s", host)
+	defaultAppConfig := b.ConfigMap.Data[defaultAppConfigFileName]
+	if defaultAppConfig == "" {
+		return nil
+	}
+	b.ConfigMap.Data[defaultAppConfigFileName] = strings.ReplaceAll(defaultAppConfig, "http://localhost:7007", baseUrl)
```

Comment:
Do we really expect only http://localhost:7007 everywhere?
It is not hardcoded in a Backstage, should we?

Reply:
This is the default value in both [upstream Backstage image](https://github.com/backstage/backstage/blob/master/packages/create-app/templates/default-app/app-config.production.yaml) and [RHDH](https://github.com/redhat-developer/rhdh/blob/main/app-config.example.production.yaml).
That's why I thought it would work when deploying Backstage/RHDH images.
But I can try to make it generic enough to update all those fields regardless.